File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int minDepth(TreeNode root) {
        if(root==null) return 0;
        int height = dfs(root);
        return height;
    }
    
    public int dfs(TreeNode node){
        //if(node==null)return 0;
        if(node.left == null && node.right == null)return 1;
        
        int leftHeight= Integer.MAX_VALUE;
        int rightHeight= Integer.MAX_VALUE;
        
        if(node.left!=null){
             leftHeight=  dfs(node.left);    
        }
        if(node.right!=null){
            rightHeight= dfs(node.right);
        }
        
        return 1+Math.min(leftHeight,rightHeight);
    }
}

File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int minDepth_01(TreeNode* node) {
        //Leafs min height Base Case 
        if(!node->left && !node->right) return 1;
        
        int leftMinH = 1e6;
        int rightMinH = 1e6;
        //Left Recursive Call of FAITH
        if(node->left) leftMinH = minDepth_01(node->left);
        
        //right Recursive Call of FAITH
        if(node->right) rightMinH = minDepth_01(node->right);
        //return this current node's Min val
        return min(leftMinH,rightMinH) + 1;
        
    }
    
    int minDepth(TreeNode* root) {
        if(!root) return 0;
        return minDepth_01(root);
    }
};